src/main/java/com/netgrif/application/engine/configuration/locale/LocaleInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package com.netgrif.application.engine.configuration.locale;
+
+import com.netgrif.application.engine.auth.domain.LoggedUser;
+import com.netgrif.application.engine.auth.service.interfaces.IUserService;
+import com.netgrif.application.engine.security.service.ISecurityContextService;
+import org.springframework.context.i18n.SimpleTimeZoneAwareLocaleContext;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.LocaleContextResolver;
+import org.springframework.web.servlet.support.RequestContextUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.time.ZoneOffset;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * The interceptor that extracts time zone from request headers.
+ * It's configuration can be found in {@link com.netgrif.application.engine.configuration.LocaleConfiguration}
+ * */
+public class LocaleInterceptor implements HandlerInterceptor {
+
+    private static final String TIMEZONE_OFFSET_HEADER_NAME = "X-Timezone-Offset";
```

Comment:
Please check if Spring provides string constant for this, otherwise I think it could be public

Reply:
I did not find any header name for timezone. However, I tried to follow custom header naming convention.
I changed this to public.